runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -378,6 +378,19 @@ class MM_ObjectAccessBarrierAPI
 		copyObjectFields(currentThread, objectClass, original, mixedObjectGetHeaderSize(), copy, mixedObjectGetHeaderSize());
 	}
 
+	VMINLINE void 
+	copyObjectFieldsToArrayElement(J9VMThread *vmThread, J9Class *arrayClazz, j9object_t srcObject, J9IndexableObject *arrayRef, U_32 index)
+	{
+		/* TODO optimizations for non-arraylet path will be added in the future */
+		vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFieldsToArrayElement(vmThread, arrayClazz, srcObject, arrayRef, index);
+	}
+	
+	VMINLINE void 
+	copyObjectFieldsFromArrayElement(J9VMThread *vmThread, J9Class *arrayClazz, j9object_t destObject, J9IndexableObject *arrayRef, U_32 index)
+	{
+		/* TODO optimizations for non-arraylet path will be added in the future */
+		vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_copyObjectFieldsFromArrayElement(vmThread, arrayClazz, destObject, arrayRef, index);
+	}
```

Comment:
in this file, APIs use UDATA for index and than downcast to I_32 if calling j9gc_ API. so let's be consistent and change index to UDATA.